runtime/compiler/runtime/MethodMetaData.c

```diff
@@ -2101,24 +2101,31 @@ UDATA getByteCodeIndexFromStackMap(J9TR_MethodMetaData * methodMetaData, void *
 
 UDATA getCurrentByteCodeIndexAndIsSameReceiver(J9TR_MethodMetaData * methodMetaData, void * stackMap, void * currentInlinedCallSite, UDATA * isSameReceiver)
    {
-   TR_ByteCodeInfo * byteCodeInfo = (TR_ByteCodeInfo *)getByteCodeInfoFromStackMap(methodMetaData, stackMap);
+   TR_ByteCodeInfo * byteCodeInfo = NULL;
 
-   if (currentInlinedCallSite)
+   if (methodMetaData && stackMap)
+      byteCodeInfo = (TR_ByteCodeInfo *)getByteCodeInfoFromStackMap(methodMetaData, stackMap);
+
+   if (isSameReceiver != 0)
+         *isSameReceiver = FALSE;
+
+   if (byteCodeInfo && currentInlinedCallSite)
       {
       void * inlinedCallSite = getFirstInlinedCallSiteWithByteCodeInfo(methodMetaData, stackMap, byteCodeInfo);
-      if (inlinedCallSite != currentInlinedCallSite)
+      if (inlinedCallSite && inlinedCallSite != currentInlinedCallSite)
          {
          void * previousInlinedCallSite;
          do
             {
             previousInlinedCallSite = inlinedCallSite;
             inlinedCallSite = getNextInlinedCallSite(methodMetaData, inlinedCallSite);
             }
-         while (inlinedCallSite != currentInlinedCallSite);
-         byteCodeInfo = (TR_ByteCodeInfo *)getByteCodeInfo(previousInlinedCallSite);
+         while (inlinedCallSite && inlinedCallSite != currentInlinedCallSite);
+         if (previousInlinedCallSite)
```

Comment:
I don't think this could ever be NULL given the logic above.  Or am I wrong?  If not, it could be removed as we don't want to be overly defensive.

Reply:
Yes, this check is not necessary. I will remove it.